src/main/java/org/opensearch/knn/index/KNNVectorScriptDocValues.java

```diff
@@ -48,10 +56,30 @@ public float[] getValue() {
         }
         try {
             BytesRef value = binaryDocValues.binaryValue();
-            ByteArrayInputStream byteStream = new ByteArrayInputStream(value.bytes, value.offset, value.length);
-            final KNNVectorSerializer vectorSerializer = KNNVectorSerializerFactory.getSerializerByStreamContent(byteStream);
-            final float[] vector = vectorSerializer.byteToFloatArray(byteStream);
-            return vector;
+            if (VectorDataType.BYTE.equals(vectorDataType)) {
+                float[] vector = new float[value.length];
+                int i = 0;
+                int j = value.offset;
+
+                while (i < value.length) {
+                    vector[i++] = value.bytes[j++];
+                }
+                return vector;
+            } else if (VectorDataType.FLOAT.equals(vectorDataType)) {
+                ByteArrayInputStream byteStream = new ByteArrayInputStream(value.bytes, value.offset, value.length);
+                final KNNVectorSerializer vectorSerializer = KNNVectorSerializerFactory.getSerializerByStreamContent(byteStream);
+                final float[] vector = vectorSerializer.byteToFloatArray(byteStream);
+                return vector;
```

Comment:
We can make this part of the enum, make `getValue` abstract method and each type provides tye specific implementation. Then on the client we try to initialize enum member and call that method, or throw exception if we cannot get any specific type

Reply:
updated